### PR TITLE
Update documentation for Navigation SDK 0.27.0

### PR DESCRIPTION
**Pull request by danesfeder**
Updates for the latest APIs in Android Navigation SDK `v0.27.0`

### CODE REVIEW

**Comment by langsmith** [comment]
Part of https://github.com/mapbox/mapbox-navigation-android/issues/1685

**Comment by langsmith** [comment]
LGTM! I pushed some small tweaks.

File: src/pages/navigation/overview/index.md [hunk 1]
```diff
@@ -146,35 +146,93 @@ NavigationRoute.builder(context)
...
+
+navigation.setLocationEngineRequest(locationEngineRequest)
+`}
+
+/>
+}}
+
+## Replaying a DirectionsRoute
+
+The SDK includes a `ReplayRouteLocationEngine`, which allows you to replay a given `DirectionsRoute` (mainly for testing, so you don't always have to code in a car 🤢). After retrieving a `DirectionsRoute`, you can create a replay engine and pass it to `MapboxNavigation`:
```
Comment (colleenmcginnis) [inline src/pages/navigation/overview/index.md]: Not to be a party 💩er, but let's remove the 🤢 emoji here.

File: src/pages/navigation/overview/off-route.md [hunk 2]
```diff
@@ -8,6 +8,14 @@ prependJs:
 
 A default off-route detection class, `OffRouteDetector`, is included in the Navigation SDK. This class uses our internal route-following library to run a series of calculations using the incoming `Location` and the current `DirectionsRoute` to determine if a user has strayed too far from the route.
 
+Upon loading a new route and if the `DirectionsRoute` `.json` is valid, route-following will start in the `RouteProgressState#INITIALIZED` state. From there, route-following will attempt to gain confidence that the GPS locations being passed to the device, are actually the location where the user is. To establish this trust, at least a few location updates need to be delivered and they must be consecutively coherent in both time and space. While it is in the process of establishing this trust, the route-following logic will report that it's still in the `RouteProgressState#INITIALIZED` state.
```
Comment (colleenmcginnis) [inline src/pages/navigation/overview/off-route.md]: >`.json`

just 

>JSON

?

File: src/pages/navigation/overview/index.md (hunk 1 above)
Comment (danesfeder) [reply src/pages/navigation/overview/index.md]: @colleenmcginnis 😁 👍

**Comment by danesfeder** [comment]
Thanks again @langsmith @colleenmcginnis 👍

**Pull request merged**